src/demo-app/datepicker/datepicker-demo.ts

```diff
@@ -40,4 +41,49 @@ export class DatepickerDemo {
 
   onDateInput = (e: MatDatepickerInputEvent<Date>) => this.lastDateInput = e.value;
   onDateChange = (e: MatDatepickerInputEvent<Date>) => this.lastDateChange = e.value;
+
+  // pass custom header component type as input
+  customHeader = CustomHeader;
+}
+
+// Custom header component for datepicker
+@Component({
+  selector: 'custom-header',
+  template: `
+      <div>
+        <button (click)="previousClicked('year')">&lt;&lt;</button>
+          <button (click)="previousClicked('month')">&lt;</button>
+        {{periodLabel}}
+        <button (click)="nextClicked('month')">&gt;</button>
+          <button (click)="nextClicked('year')">&gt;&gt;</button>
+      </div>
+  `
+})
+export class CustomHeader {
+  constructor(@Host() public calendar: MatCalendar<any>,
+              private _dateAdapter: DateAdapter<any>) {}
+
+  get periodLabel() {
+    let year = this._dateAdapter.getYearName(this.calendar.activeDate);
+    let month = (this._dateAdapter.getMonth(this.calendar.activeDate) + 1);
+    return `${month}/${year}`;
+  }
+
+  previousClicked(mode: 'month' | 'year') {
+    this.calendar.activeDate = mode == 'month' ?
+            this._dateAdapter.addCalendarMonths(this.calendar.activeDate, -1) :
+            this._dateAdapter.addCalendarYears(
```

Comment:
can probably collapse to 1 line after correcting indentation

Reply:
done in 6d0a4d2264e085c011770bf3de0fe381598dbe33